Adds GDAL Vector drivers description (#6123)
qgis/QGIS-Documentation: docs/user_manual/introduction/qgis_configuration.rst

> `GDAL <https://gdal.org>`_ is a data exchange library for vector and raster
> files. It provides drivers to read and or write data in different formats.

Reviewer: ```suggestion
`GDAL <https://gdal.org>`_ is a data exchange library sfor geospatial data that supports a
large number of vector and raster formats.
It provides drivers to read and (often) write data in these formats.
```

> The :guilabel:`GDAL` tab currently exposes the drivers for raster and vector

Reviewer: ```suggestion
The :guilabel:`GDAL` tab exposes the drivers for raster and vector
```

formats with their capabilities.

> GDAL driver options

Reviewer 2: ```suggestion
Raster driver options
```

     Sample of Pyramids profile

GDAL raster and vector drivers
..............................

> The :guilabel:`Raster Drivers` and :guilabel:`Vector Drivers` (in a separated
> tab) allow you to define which GDAL driver is enabled to read and/or
> write files, as in some cases more than one GDAL driver is available.

Reviewer: ```suggestion
The :guilabel:`Raster Drivers` and :guilabel:`Vector Drivers` (in separate tabs)
allow you to configure the GDAL drivers.
```
Not sure what can and can not do here,but I can't find a  way to enable / disable.

Author: Before each driver, there's a checkbox. You can disable each of the drivers.

Reviewer: > Before each driver, there's a checkbox. You can disable each of the drivers.

Thanks ( for some reason, I did not notice the checkboxes).
I have now updated the suggestion.

Author: @havatv I am not so sure about your suggestion. It removes the explanation of why someone would want to disable a driver. Besides, you say you can configure the GDAL drivers, but there isn't much you can configure. Unless you talking about the double-click trick for rasters.

Reviewer: > @havatv I am not so sure about your suggestion. It removes the explanation of why someone would want to disable a driver. Besides, you say you can configure the GDAL drivers, but there isn't much you can configure. Unless you talking about the double-click trick for rasters.

OK. But what about *Edit Create Options* / *Edit Pyramids  Options* (3.14)

Author: > > @havatv I am not so sure about your suggestion. It removes the explanation of why someone would want to disable a driver. Besides, you say you can configure the GDAL drivers, but there isn't much you can configure. Unless you talking about the double-click trick for rasters.
> 
> OK. But what about _Edit Create Options_ / _Edit Pyramids Options_ (3.14)

That is described in the previous section. This text is about the Raster drivers and Vector Drivers lists.
